### ISSUE TITLE
How to use the yolo-world trained model to extract features for simple classification of downstream tasks, such as training LR models.

### ISSUE THREAD

**Issue by Meicsu199345** [AILab-CVC/YOLO-World#397]
Background
Continuing from the previous question, under the guidance of the author, we successfully applied yolo-world as a prompt, prompt like “person holding small white cigarette and smoking cigarette with mouth”, work excellent in our work.

Question
Now, we have a new question, can we use yolo-world trained model as base, providing embedding to fine-grained differentiation of samples for downstream tasks. For example, two categories: white cigarettes and black cigarettes.

Related work
Previously, clip was used to extract features, and a linear regression model was used downstream to distinguish them. This eliminates the need for retraining and does not affect other labels. Added fine-grained classification capabilities; 

I learn yolo-world network backbone-neck-head，should i extra neck featuremap as output?

thank you all for your answers. I wish yolo-world will get better and better.
link：
https://github.com/openai/CLIP?tab=readme-ov-file#linear-probe-evaluation
![img_v3_02bv_76b89675-6168-4f2a-b10a-92f828ef2aeg](https://github.com/AILab-CVC/YOLO-World/assets/16251955/60e8d835-1b1b-46f1-b85f-83a4d62b65dc)
`
code like
import os
import clip
import torch

import numpy as np
from sklearn.linear_model import LogisticRegression
from torch.utils.data import DataLoader
from torchvision.datasets import CIFAR100
from tqdm import tqdm

# Load the model
device = "cuda" if torch.cuda.is_available() else "cpu"
model, preprocess = clip.load('ViT-B/32', device)

# Load the dataset
root = os.path.expanduser("~/.cache")
train = CIFAR100(root, download=True, train=True, transform=preprocess)
test = CIFAR100(root, download=True, train=False, transform=preprocess)

def get_features(dataset):
    all_features = []
    all_labels = []
    
    with torch.no_grad():
        for images, labels in tqdm(DataLoader(dataset, batch_size=100)):
            features = model.encode_image(images.to(device))

            all_features.append(features)
            all_labels.append(labels)

    return torch.cat(all_features).cpu().numpy(), torch.cat(all_labels).cpu().numpy()

# Calculate the image features
train_features, train_labels = get_features(train)
test_features, test_labels = get_features(test)

# Perform logistic regression
classifier = LogisticRegression(random_state=0, C=0.316, max_iter=1000, verbose=1)
classifier.fit(train_features, train_labels)

# Evaluate using the logistic regression classifier
predictions = classifier.predict(test_features)
accuracy = np.mean((test_labels == predictions).astype(float)) * 100.
print(f"Accuracy = {accuracy:.3f}")
`
